var/spack/repos/builtin/packages/arborx/package.py

```diff
@@ -80,13 +80,16 @@ class Arborx(CMakePackage, CudaPackage, ROCmPackage):
 
     for arch in CudaPackage.cuda_arch_values:
         cuda_dep = "+cuda cuda_arch={0}".format(arch)
-        depends_on("kokkos {0}".format(cuda_dep), when=cuda_dep)
+        depends_on("kokkos {0}".format(cuda_dep), when="~trilinos {}".format(cuda_dep))
+        depends_on("trilinos {0}".format(cuda_dep), when="+trilinos {}".format(cuda_dep))
 
     for arch in ROCmPackage.amdgpu_targets:
         rocm_dep = "+rocm amdgpu_target={0}".format(arch)
-        depends_on("kokkos {0}".format(rocm_dep), when=rocm_dep)
+        depends_on("kokkos {0}".format(rocm_dep), when="~trilinos {}".format(rocm_dep))
+        depends_on("trilinos {0}".format(rocm_dep), when="+trilinos {}".format(rocm_dep))
 
```

Comment:
FWIW. We are switching the core to use f-strings (https://docs.python.org/3/tutorial/inputoutput.html) for string formatting as we make changes so it's something you might want to consider now or in the future.